internal/registry/service.go

```diff
@@ -248,11 +245,7 @@ func (client *Client) GetIterationFromChannel(
 	}
 
 	if resp.Payload.Channel != nil {
-		if resp.Payload.Channel.Iteration != nil {
-			return resp.Payload.Channel.Iteration, nil
-		}
-		return nil, fmt.Errorf("there is no iteration associated with the channel %s",
-			channelName)
+		return resp.Payload.Channel, nil
 	}
 
 	return nil, fmt.Errorf("there is no channel with the name %s associated with the bucket %s",
```

Comment:
nit: reverse so that this is the happy path

```
if resp.Payload.Channel == nil {
    	return nil, fmt.Errorf("there is no channel with the name %s associated with the bucket %s", channelName, bucketSlug)
}

return resp.Payload.Channel, nil
```